test/ts/prect-test.tsx

```diff
@@ -1,32 +1,425 @@
-import { h, render, Component } from 'preact';
+import { h, render, Component, SFC, ChildContextProvider, ComponentProps, ComponentClass, VNode } from 'preact';
 
-interface DummyProps {
-	initialInput: string;
-}
+// import 'preact/devtools';
 
-interface DummyState {
-	input: string;
-}
+// #Components and StatellesComponents
+type DummyProps = { initialInput: string; }
+type DummyState = { input: string; }
 
 class DummyComponent extends Component<DummyProps, DummyState> {
+	// static members cannot be typechecked by inferenece
+	static defaultProps: DummyProps = {
+		initialInput: 'hmm'
+	}
 	constructor(props: DummyProps) {
 		super(props);
 		this.state = {
 			input: `x${this.props}x`
 		}
 	}
 
-	render({ initialInput }: DummyProps, { input }: DummyState) {
-		return <DummerComponent initialInput={initialInput} input={input} />
+	render(props: DummyProps, state: DummyState) {
+		const { initialInput } = this.props;
+		const { input } = this.state;
+		return (
+			<DummerComponent
+				initialInput={initialInput}
+				input={input}
+			/>
+		);
 	}
 }
 
-interface DummerComponentProps extends DummyProps, DummyState {
-
-}
+type DummerComponentProps = DummyProps & DummyState;
 
 function DummerComponent({ input, initialInput }: DummerComponentProps) {
 	return <div>Input: {input}, initial: {initialInput}</div>;
 }
+DummerComponent.defaultProps = { input: 'yay' };
+DummerComponent.displayName = 'Hello';
+
+const DummerComponent2: SFC<DummerComponentProps> = ({ initialInput, input }) => (
+	<div>Input: {input}, initial: {initialInput}</div>
+);
+DummerComponent2.defaultProps = { input: 'yay' };
+DummerComponent2.displayName = 'Hello2';
+
+// Children Projection
+
+type LayoutProps = {
+	children?: [LayoutProjecitionFn]
```

Comment:
The typo here might also cause some dev headaches. Should be `LayoutProjectionFn`.